lib/gem/specification.rb

```diff
@@ -3,21 +3,60 @@ class Specification
     alias __licenses licenses
 
     LICENSE_REFERENCES = [
-      /released under the (?<l>[\s\w]*) license/i,
-      /same license as (?<l>[\s\w]*)/i,
-      /^(?<l>[\s\w]*) License, see/i,
-      /^(?<l>[\w]*) license$/i,
-      /\(the (?<l>[\s\w]*) license\)/i,
-      /^license: (?<l>[\s\w]*)/i,
-      /^released under the (?<l>[\s\w]*) license/i,
-      /license: (?<l>[\s\w]*)$/i,
-      /^same as (?<l>[\s\w]*)/i,
-      /license of (?<l>[\s\w]*)/i
+      /released under the (?<l>[\s\w]+) license/i,
+      /same license as (?<l>[\s\w]+)/i,
+      /same terms of (.+)/i,
+      /^(?<l>[\s\w]+) License, see/i,
+      /^(?<l>[\w]+) license$/i,
+      /\(the (?<l>[\s\w]+) license\)/i,
+      /^license: (?<l>[\s\w]+)/i,
+      /^released under the (?<l>[\s\w]+) license/i,
+      /license: (?<l>[\s\w]+)$/i,
+      /^same as (?<l>[\s\w]+)/i,
+      /license of (?<l>[\s\w]+)/i
     ].freeze
 
     def licenses
-      ary = (__licenses || []).keep_if { |l| !l.empty? }
-      ary.empty? ? guess_licenses : ary
+      cleaned_up_licenses
+    end
+
+    def cleaned_up_licenses
+      before = gem_or_file_license
+
+      log_warning "License #{before.inspect}"
+
+      # manually clean up some cruft
+      after = before.collect do |license|
+        stringified = license.to_s
```

Comment:
Nitpicking, but this could be a `when` with all three values an `license.to.s_upcase`.